### PR TITLE
Implement spending of entire UTxOs using build command with no change outputs

### LINKED ISSUE #3041: [BUG] - transaction build attempts to create UTxO with 0 lovelace as change
**Internal/External**
*Internal*   if an IOHK staff member.

**Area**
*Other*   Any other topic (Delegation, Ranking, ...).

**Summary**
When I use `transaction build` and balance the outputs so that there is no change then the cli still attempts to create a UTxO which fails due to minimum constraint. If there's no remainder it should skip trying to create UTxO for change of 0 lovelace.

**Steps to reproduce**
1. Have some ada
```
0c58e04f797e1fbcc4c26730af010d24d74a1df11631363c279972654da3305d     1        10000000 lovelace + TxOutDatumHashNone
```
2. Use `transaction build` to send 10 ada minus the exact fee of 167217 lovelace
```
$ CARDANO_NODE_SOCKET_PATH=node/node.sock cardano-cli transaction build --tx-in 0c58e04f797e1fbcc4c26730af010d24d74a1df11631363c279972654da3305d#1 --tx-out "$(<AlwaysSucceeds.addr)+9832783" --tx-out-datum-hash 9e1199a988ba72ffd6e9c269cadb3b53b5f360ff99f112d9b2ee30c4d74ad88b --change-address $(<acct.addr) --out-file txbody --alonzo-era --testnet-magic $(<node/db/protocolMagicId)
```
3. Observe error
```
Command failed: transaction build  Error: The transaction does balance in its use of ada, however the net balance does not meet the minimum UTxO threshold. 
Balance: Lovelace 0
Offending output (change output): addr_test1vpl95kfrrvx6uxd54a9vwjw7nrv88g0euq70jn5jy8dlysshlxfrw + 
Minimum UTxO threshold: Lovelace 999978
The usual solution is to provide more inputs, or inputs with more ada to meet the minimum UTxO threshold
```

**Expected behavior**

**System info (please complete the following information):**
- OS Name: Mint
- OS Version 20.2 (Uma)
- Node version: cardano-node 1.28.0 - linux-x86_64 - ghc-8.10 git rev 708de685d49ec6af4b2d8b3cbfa0eca0e9e43edf
- CLI version: cardano-cli 1.28.0 - linux-x86_64 - ghc-8.10 git rev 708de685d49ec6af4b2d8b3cbfa0eca0e9e43edf

### PR DESCRIPTION
**Pull request by Jimbo4350**
Resolves #3041

### CODE REVIEW

**Review by dcoutts — COMMENTED** [review]
Does this actually work? Doesn't the balance check already fail if we have a balance that is less than the min utxo (which of course includes zero). 

Note also that even if this passes the balance check, it does not involve paying any less in fees for the case where the change comes out to be zero. We're still calculating the fees assuming we'll have a change output. That's a much harder one to solve (and why I didn't bother in the first place).

**Comment by Jimbo4350** [comment]
> Does this actually work?

I forgot to push a local change that modifies `balanceCheck` so this works now. I'm not going to bother to go through the hassle of reducing the fee calculation because I agree its not worth it.

File: cardano-api/src/Cardano/Api/Fees.hs [hunk 1]
```diff
@@ -922,8 +924,18 @@ makeTransactionBodyAutoBalance eraInMode systemstart history pparams
    era' :: CardanoEra era
    era' = cardanoEra
 
+   -- In the event of spending the exact amount of lovelace in
+   -- the specified input(s), this function excludes the change
+   -- output.
```
Comment (dcoutts) [inline cardano-api/src/Cardano/Api/Fees.hs:929]: Can we also comment that this does not save any fees. This is a subtle point so it's worth a proper comment explaining what is going on.

**Review by dcoutts — APPROVED** [review]
LGTM, though as suggested below I think it's worth explaining properly in the code comments.

**Pull request merged**